### ISSUE TITLE
Decouple building uses from code (ADMIN, SR, etc.)

### ISSUE THREAD

**Issue by daren-thomas** [architecture-building-systems/CEAforArcGIS#73]
Currently, we use `GlobalVariables.list_uses` to define the list of valid uses for the program. These uses are linked with the schedules and with the attributes in `buildings.shp`. While we try to make sure (e.g. in `properties.py`) that we don't reference this list explicitly, we still make assumptions about them. See in `properties.py`:

```python
 if generate_uses:
        uses_df = pd.DataFrame({'Name': name, 'ADMIN': ADMIN, 'SR': SR,
                                'REST': REST, 'RESTS': RESTS, 'DEPO': DEPO,
                                'COM': COM, 'MDU': MDU, 'SDU': SDU,
                                'EDU': EDU, 'CR': CR, 'HEALTH': HEALTH,
                                'SPORT': SPORT, 'SWIM': SWIM, 'PUBLIC': PUBLIC,
                                'SUPER': SUPER, 'ICE': ICE, 'HOT': HOT,
                                'INDUS': INDUS})
```

As an example. If we change `GlobalVariables.list_uses`, then we need to change this list (and others, like the local variables `ADMIN`, `SR` etc.. And also change the list of schedule files.

I propose to calculate `list_uses` based on the available schedules for the location (see #36) and move the local variables to a dictionary, so that they are not hard-coded anymore!

**NOTE:** The use "DEPO" has special treatment!

**Comment by JIMENOFONSECA** [comment]
@daren-thomas is this still necessary?

**Comment by daren-thomas** [comment]
@JIMENOFONSECA I would need to take a closer look. I gather you just added a new use (LIBRARY) - how did that go? If all you needed to do was change the archetypes excel files then this issue can be closed. However, since the current master branch is failing with a `XLRDError: No sheet named <u'LIBRARY'>` I believe we are not quite there yet...

**Comment by JIMENOFONSECA** [comment]
Yep the next pr in the pipeline fixes this.but it is good that this error exists that means he forgot to add a tab to the archetypes folder. Should we create an error messagge?

Best, Jimeno

On 20 Mar 2017, at 21:51, Daren Thomas <notifications@github.com<mailto:notifications@github.com>> wrote:


@JIMENOFONSECA<https://github.com/JIMENOFONSECA> I would need to take a closer look. I gather you just added a new use (LIBRARY) - how did that go? If all you needed to do was change the archetypes excel files then this issue can be closed. However, since the current master branch is failing with a XLRDError: No sheet named <u'LIBRARY'> I believe we are not quite there yet...
